### ISSUE TITLE
New linter: detect empty destructuring bindings

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2960]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

An empty vector or map used as a destructuring binding binds no values and is usually residue from a refactor.

```clojure
(let [{} value] value)
(let [[[]] values] values)
```

An empty function argument vector is normal and must not warn.

**Describe the solution you'd like**

Add an `:empty-binding` linter, enabled at warning level by default, that reports empty sequential and map destructuring forms while allowing an empty function argument vector. Nested empty destructuring should still be detected.

**Describe alternatives you've considered**

The existing unsupported-binding checks could reject these forms, but they are valid Clojure. Treating this as an unused binding also loses the specific structural problem.

**Additional context**

I am interested in submitting a focused PR with nested destructuring and function argument tests if this approach is accepted.

**Comment by NoahTheDuke** [comment]
Is this a frequent issue?

**Comment by marco-carvalho** [comment]
The pinned Metabase regression corpus contains 39 occurrences. Looking at a sample, many are intentional placeholders in macro DSLs such as `mt/with-temp` and `mt/with-actions`, where `{}` means the created value is ignored. So the syntax is frequent, but the corpus evidence suggests the bug signal is weaker and noisier than I expected. That argues against enabling it by default unless the check can distinguish those binding DSLs.
